app/services/profile_picture.rb

```diff
@@ -1,30 +1,57 @@
 # frozen_string_literal: true
 
+include CloudinaryHelper
+
 class ProfilePicture
   DEFAULT_AVATAR = '/assets/default_ifme_avatar.png'
+  # Future task: Optimize Cloudinary image size based on responsive layout
+  DEFAULT_SIZE = 150
+  # JavaScript that allows image to fallback if main src URL cannot load
+  DEFAULT_AVATAR_JS = "this.onerror=null;this.src='#{DEFAULT_AVATAR}'"
 
   class << self
-    def fetch(avatar, class_name)
-      html = "<div class='#{class_name}' " \
-             "style='background: url(#{avatar_url(avatar)})'></div>"
+    # If the image is from Cloudinary, use cl_image_tag
+    # Otherwise, in production, use Cloudinary remote fetching
+    # to grab the image from the if-me.org domain.
+    # In development environment, don't use Cloudinary remote fetching.
+    def fetch(path, class_name)
+      html = if cloudinary(path)
+               cl_img(get_cloudinary_image_id(path), true, class_name)
+             elsif path.present? && Rails.env.production?
+               cl_img(full_url(path), false, class_name)
+             else
+               html_img(path, class_name)
+             end
       html.html_safe
     end
 
     private
 
-    def avatar_url(avatar)
-      if avatar.blank?
-        DEFAULT_AVATAR
-      elsif avatar.include?('/assets/contributors/')
-        avatar
-      else
-        response = Net::HTTP.get_response(URI.parse(avatar))
-        valid_response?(response) ? avatar : DEFAULT_AVATAR
-      end
+    def cloudinary(path)
+      path&.include?('.cloudinary.com')
+    end
+
+    def get_cloudinary_image_id(path)
+      path.split('/').last.split('.').first
+    end
+
+    def full_url(path)
+      "https://#{primary_domain}#{path}"
```

Comment:
Where is primary_domain coming from? I don't see it passed as an argument. Also can't you get the url from `request.url`?

Reply:
production.rb had it but, I thought when I initially tested it it was accessible; I guess not. I'm updating the development.rb file to look similar to production.rb (as @HashNotAdam has done), and changing the accessing of the config value to `Rails.application.config.action_controller.asset_host` (which seems to work).